lib/config/applyrule-targeted.cpp

```diff
@@ -63,21 +63,21 @@ const std::vector<ApplyRule::Ptr>& ApplyRule::GetTargetedServiceRules(const Type
 bool ApplyRule::AddTargetedRule(const ApplyRule::Ptr& rule, const String& targetType, ApplyRule::PerSourceType& rules)
 {
 	if (targetType == "Host") {
-		std::vector<const String *> hosts;
+		std::set<String> hosts;
 
 		if (GetTargetHosts(rule->m_Filter.get(), hosts)) {
-			for (auto host : hosts) {
-				rules.Targeted[*host].ForHost.emplace_back(rule);
+			for (const auto& host : hosts) {
+				rules.Targeted[host].ForHost.emplace_back(rule);
 			}
 
 			return true;
 		}
 	} else if (targetType == "Service") {
-		std::vector<std::pair<const String *, const String *>> services;
+		std::set<std::pair<const String, const String>> services;
```

Comment:
Same with the `const` here.

Maybe for some context here: previously, the `const` was correct (and maybe even needed) here because it didn't make the elements in the vector `const`, but rather it was a (non-const) pointer to a `const String`. (Yes, that syntax is strange.)